[MIG] backport peppol modules to 14.0

benwillig opened the pull request.

benwillig committed:
[MIG] backport peppol modules to 14.0

benwillig committed:
[IMP] account_peppol: Check 0208 and 9925 + remove warnings (aae04364163aa9ff95d59c608a49ecffd06e6d4b)

sbidoul committed:
Fix versions in manifest

sbidoul committed:
Unformat account_edi_ubl_cii_tax_extension to keep it comparable with upstream

sbidoul committed:
Apply pre-commit auto fixes

sbidoul committed:
Test with python 3.8

The code is not compatible with 3.6
at the moment because of the
walrus operator.

sbidoul committed:
Fix typing issue

sbidoul commented:
@benwillig thanks for this!

I made a few tweeks so pre-commit passes and updated the CI to test with python 3.8 as 3.6 compatibility is not necessary I think.

There are a couple of remaining test failures. Could you have a look?

huguesdk committed:
[FIX] neutralize by using demo mode

use demo mode instead of test mode when neutralizing. in test mode,
documents will try to be sent and received on the peppol test network
using the credentials of the live one, which will fail. the test network
requires a separate registration.

this is the same behavior as in odoo 17.0.

benwillig committed:
[FIX] fixed account_edi_ubl_cii_tax_extension and proxy user constraint length

sbidoul commented:
@benwillig there are still a few errors creating constraints.
